smartgymapi/handlers/sport_schedule.py

```diff
@@ -1,32 +1,44 @@
 import logging
+
 from marshmallow import ValidationError
 from pyramid.httpexceptions import HTTPBadRequest, HTTPInternalServerError
 from pyramid.view import view_defaults, view_config
-from smartgymapi.models import persist, rollback, commit, delete
+
 from smartgymapi.lib.factories.sport_schedule import SportScheduleFactory
 from smartgymapi.lib.validation.sport_scheme import SportScheduleSchema
-from smartgymapi.models.sport_schedule import SportSchedule
+from smartgymapi.models import persist, rollback, commit, delete
+from smartgymapi.models.sport_schedule import SportSchedule, get_sport_schedule_by_name
 
 log = logging.getLogger(__name__)
 
 
 @view_defaults(containment=SportScheduleFactory,
-               permission='public',
+               permission='sport_schedule',
                renderer='json')
 class RESTSportScheme(object):
     def __init__(self, request):
         self.request = request
 
     @view_config(context=SportScheduleFactory, request_method="GET")
     def list(self):
-        return SportScheduleSchema(many=True).dump(self.request.context.get_sport_schedules())
+        try:
+            return SportScheduleSchema(many=True).dump(
+                self.request.context.get_sport_schedules()).data
+        except ValidationError as e:
+            raise HTTPBadRequest(json={'message': str(e)})
 
     @view_config(context=SportSchedule, request_method="GET")
     def get(self):
-        return SportScheduleSchema().dump(self.request.context)
+        return SportScheduleSchema().dump(self.request.context).data
 
     @view_config(context=SportScheduleFactory, request_method="POST")
     def post(self):
+        try:
+            if get_sport_schedule_by_name(self.request.json_body['name']):
+                raise HTTPBadRequest(json={'message': 'Sport schedule name already exists'})
+        except KeyError as e:
+            raise HTTPBadRequest(json={'message': str(e)})
```

Comment:
Eigenlijk checkt hij of in de json body een name heeft dan gaat hij hem alvast checken, daarna valideerd hij het hele model. Maar het refactoren naar aparte methode doe ik denk later